pkg/controller/syncconfig/utils.go

```diff
@@ -0,0 +1,113 @@
+package syncconfig
+
+import (
+	"context"
+	"strings"
+
+	templateapi "github.com/openshift/origin/pkg/template/apis/template"
+	"github.com/openshift/origin/pkg/template/generator"
+	"github.com/openshift/origin/pkg/template/templateprocessing"
+	"github.com/vshn/espejo/pkg/apis/sync/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func recreateOject(ctx context.Context, unstructObj *unstructured.Unstructured, c client.Client) error {
```

Comment:
Depending on the use case, using delete and create can have some undesired side effects. The delete operation can result in a cascading/chaining where other objects get deleted as well. Consider to use something like update/replace if it exists (the CLI has a replace feature).

Reply:
That's why I use the `forceRecreate` flag. The problem is that some fields of K8s objects are immutable (i.e. service ClusterIPs), so we have to delete and recreate them.